src/plain.c

```diff
@@ -1,5 +1,75 @@
 #include "h2get.h"
+#include <errno.h>
 #include <unistd.h>
+#include <poll.h>
+
+static int wait_for_read(int fd, int tout)
+{
+    struct pollfd fds[1];
+    fds[0].fd = fd;
+    fds[0].events = POLLIN;
+    return poll(fds, 1, tout);
+}
+
+static int plain_accept(struct h2get_conn *listener, struct h2get_conn *conn, int tout)
+{
+    if (listener->fd < 0) {
+        return -1;
+    }
+
+    int r;
+    while ((r = wait_for_read(listener->fd, tout)) < 0 && errno == EINTR) {}
+    if (r == 0) {
+        return H2GET_ERROR_TIMEOUT;
+    }
+
+    conn->sa.len = sizeof(conn->sa.sa_storage);
+    conn->fd = accept(listener->fd, (void *)&conn->sa.sa_storage, &conn->sa.len);
+    if (conn->fd < 0) {
+        return -1;
+    }
+    conn->sa.sa = (void *)&conn->sa.sa_storage;
+    conn->ops = listener->ops;
+    conn->state = H2GET_CONN_STATE_CONNECT;
+
+    return 0;
+}
+
+static int plain_write(struct h2get_conn *conn, struct h2get_buf *bufs, size_t nr_bufs)
+{
+    int ret;
+    size_t i;
+    for (i = 0; i < nr_bufs; i++) {
+        int wlen = 0;
+        do {
+            ret = write(conn->fd, bufs[i].buf + wlen, bufs[i].len - wlen);
+            if (ret < 0) {
+                return -1;
+            }
+            wlen -= ret;
+        } while (wlen > 0);
```

Comment:
I know this is from ssl.c so this may be out of this PR's scope, but I don't see what this code is doing. `wlen` must be negative value so this do while loop is meaningless?

```c
do {
    ret = write(conn->fd, bufs[i].buf + wlen, bufs[i].len - wlen);
    if (ret < 0) {
        return -1;
    }
    wlen += ret;
} while (wlen < bufs[i].len);
```
does make sense for me.